plugin/rewrite/name.go

```diff
@@ -16,6 +18,36 @@ func newNameRule(from, to string) (Rule, error) {
 
 // Rewrite rewrites the the current request.
 func (rule *nameRule) Rewrite(w dns.ResponseWriter, r *dns.Msg) Result {
+	rname := r.Question[0].Name
+	// Rewrite based on substring
+	if strings.Contains(rule.From, "sub:") {
```

Comment:
Shouldn't this be `strings.HasPrefix`?

Reply:
@miekg , for the context, the `sub:` was born prior to `regex:`. In its current state `rewrite` requires full match. The `sub:` allows to match and replace a substring in a name. For example, I want to replace `contoso.com` with `service.consul`. I would do it with `sub:contoso.com service.consul` and no regular expressions.

>  Shouldn't this be `strings.HasPrefix`?

Agree. 👍 will re-work